datafusion/sql/src/unparser/expr.rs

```diff
@@ -30,10 +34,38 @@ use sqlparser::ast::{
 
 use super::Unparser;
 
+/// DataFusion's Exprs can represent either an `Expr` or an `OrderByExpr`
+pub enum Unparsed {
+    // SQL Expression
+    Expr(ast::Expr),
+    // SQL ORDER BY expression (e.g. `col ASC NULLS FIRST`)
+    OrderByExpr(ast::OrderByExpr),
+}
+
+impl Unparsed {
+    fn into_order_by_expr(self) -> Result<ast::OrderByExpr> {
```

Comment:
Maybe we can call this `try_into_order_by_expr` to reflect the fact it is fallable

Perhaps it should also be marked `pub` 🤔